source/dagger/gameplay/team_game/player_controller.cpp

```diff
@@ -10,11 +11,40 @@
 #include "core/graphics/window.h"
 #include "gameplay/common/simple_collisions.h"
 
-#include "core/game/transforms.h"
+
 
 using namespace dagger;
 using namespace lab;
 
+void lab::GenerateRoom(int idNext_,lab::NextLvl& currentLvl_, Transform &tr_)
+{
+    if(currentLvl_.id==0)
```

Comment:
This can be moved to data; have a file with lines like:

1 tilemaps/lab/lab.map
2 tilemaps/lab/hallway.map

Load that into a map, and then just send a request from there instead of accumulating branches like this.